lib/controller.js

```diff
@@ -641,6 +663,213 @@ controller.tesselFirmwareVerion = function(opts) {
+      // (Terminal freezes imediatelly after poweroff the Tessel)
+
+      ch.once('error', function(e) {
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
+          reject();
+        } else {
+          logs.warn('Error while connected to ' + tessels[id].lanConnection.host + ':', e);
+          reject(e);
+        }
+
+      });
+      ch.once('exit', function(e) {
+
+        if (e === 255) {
+          controller.menu.clear();
+          controller.ssh.notAuthorized();
+          reject();
+        } else if (e === 127) { // exit by user
+          // clear console
+          controller.menu.clear();
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed!\n');
+          resolve();
+        } else if (e === 0) {
+          // everything works fine ... now lets clear the terminal
+          controller.menu.clear();
+          logs.info('Connection to ' + tessels[id].lanConnection.host + ' closed!\n');
+          resolve();
+        } else {
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed due to reason:\n', e);
+          reject(e);
+        }
+      });
+
+      ch.once('close', function(e) {
+        // tessel powers off
+        if (e === 0) {
+          // everything works fine...
+          resolve();
+        } else if (e === 255) {
+          //reject();
+          process.exit();
+        } else if (e === 127) {
+          reject();
+        } else {
+          logs.warn('Connection to ' + tessels[id].lanConnection.host + ' closed due to reason:\n', e);
+          reject();
+        }
+      });
+    }
+  },
+  notAuthorized: function() {
+    logs.warn('Sorry, you are not authorized!');
+    logs.info('Maybe this Tessel didn\'t got "t2 provision" until now?');
+  }
+};
+
+/*
+  The T2 root command is used to login into the firmware and gaining superuser access.
+  The security is provided by RSA - your Tessel get the keys while provisioning via USB.
+  
+  If you have only one Tessel, the T2 root command will login directly else there is 
+  a ncurses like menu you can select the Tessel you like to gain root access...
+
+  The structure of code and maybe unusual parts are paid due being testable.
+  The lightweight terminal-menu isn't written testable what needs a little bit hacking.
+
+  Finally some parts are causing from my personnal learning curve about writing unit tests using sinon!
+*/
+controller.root = function(opts) {
+  // ~ conversion to home because spawn isn't able to handle this right
+  if (opts.path && opts.path.substring(0, 1) === '~') {
+    var home = process.env[(process.platform === 'win32') ? 'USERPROFILE' : 'HOME'];
+    if (/^~/.test(opts.path)) {
+      opts.path = opts.path.replace(/~/, home);
+    }
+  }
+  opts.root = true;
+  return new Promise(function(resolve, reject) {
+    controller.menu.seek(opts)
```

Comment:
Additionally, if you need to make sure that you get a `lan` Tessel, you can set the `--lan` flag to true.